tests/rptest/tests/compatibility/arroyo_test.py

```diff
@@ -27,30 +28,34 @@ def __init__(self, ctx, *args, **kwargs):
                          *args,
                          **kwargs)
 
+    @ok_to_fail  # https://github.com/redpanda-data/redpanda/issues/6826
     @cluster(num_nodes=4)
     def test_arroyo_test_suite(self):
         test_node = self.preallocated_nodes[0]
 
         try:
             env_preamble = f"DEFAULT_BROKERS={self.redpanda.brokers()}"
 
-            failed = False
-            for line in test_node.account.ssh_capture(
-                    f"{env_preamble} "
-                    f"python3 -m pytest {ArroyoTest.TEST_SUITE_PATH} "
-                    "-k KafkaStreamsTestCase -rf",
-                    combine_stderr=True,
-                    timeout_sec=120):
-                self.logger.info(line)
-                if 'FAILED' in line:
-                    failed = True
-
-            if failed:
+            # Setting allow_fail to false so in the case pytest errors, the code
+            # flow continues as normal and logs the test output
+            arroyo_logs = test_node.account.ssh_capture(
+                f"{env_preamble} "
+                f"python3 -m pytest {ArroyoTest.TEST_SUITE_PATH} "
+                "-k KafkaStreamsTestCase -rf",
+                combine_stderr=True,
+                allow_fail=False,
+                timeout_sec=120)
+
+            # Log output from arroyo
+            self.logger.info("\n".join(arroyo_logs))
+
+            # If any errors detected raise error
+            if any(['FAILED' in line for line in arroyo_logs]):
```

Comment:
`arroyo_logs` is an iterator.  It is drained on line 50 when you stringize it.  When you try to use it again here, you're not getting anything at all.  I think the previous iteration of this where we kept the orginal for loop was fine

Reply:
nice find